spec/forms/pages/conditions_form_spec.rb

```diff
@@ -53,27 +62,37 @@
     end
   end
 
-  describe "#update" do
+  describe "#update_condition" do
     context "when validation pass" do
       let(:condition) { Condition.new id: 3, form_id: 1, page_id: 2, routing_page_id: 1, check_page_id: 1, answer_value: "Wales", goto_page_id: 3 }
 
-      it "updates a condition" do
+      before do
         ActiveResource::HttpMock.respond_to do |mock|
           mock.post "/api/v1/forms/1/pages/2/conditions", post_headers, { success: true }.to_json, 200
           mock.put "/api/v1/forms/1/pages/2/conditions/3", post_headers, { success: true }.to_json, 200
         end
 
         conditions_form.answer_value = "England"
         conditions_form.goto_page_id = 4
+      end
+
+      it "calls assign_skip_to_end" do
+        allow(conditions_form).to receive(:assign_skip_to_end)
 
-        expect(conditions_form.update).to be_truthy
+        conditions_form.update_condition
+
+        expect(conditions_form).to have_received(:assign_skip_to_end).exactly(1).times
+      end
```

Comment:
I think this is testing implementation details rather than behaviour. What is the expected behaviour we expecting to happen when submit is called with specific params passed in.

Reply:
Ok, I see you have united tested the `assign_skip_to_end`.